sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/CompressedSource.java

```diff
@@ -319,6 +320,13 @@ public final boolean producesSortedKeys(PipelineOptions options) throws Exceptio
     return sourceDelegate.producesSortedKeys(options);
   }
 
+  @Override
+  public void populateDisplayData(DisplayData.Builder builder) {
+    builder
+        .include(sourceDelegate)
+        .add("compressionMode", channelFactory.toString());
```

Comment:
not sure `.toString()` is right here. Will work for the ENUMs, but will not work well for `DecompressAccordingToFilename`. Maybe use something like `getClass().getSimpleName()`?